src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -333,10 +345,7 @@ private[io] object StageWriter {
                                            fileUploadResults: List[FileUploadResult])
   : Unit = {
     val table = params.table.get
-    val tempTable =
```

Comment:
Can you add a parameter to disable this new behavior? My fear is that some customers are using the old staging table name scheme.

Reply:
After more consideration, the regression about item-4 should not exist. If the schema name is MY_SCHEMA, the table rename or swap when committing the WRITE will fail.

Reply:
I don't understand the concern here. We are creating this temp table, how can customers be using it ?

Reply:
@sfc-gh-sshankar The staging table only exists when writing a data frame to snowflake. But the table is a normal table (not a snowflake TEMP table). So it could be seen by others users when it exists.
I think it is extremely low possibility that the customers depends on the staging table name. In theory, it still is possible. (The example I can imagine is that the customer may clean up some unused staging tables like "%_staging_%", In most cases, if the writing job fails, the staging table will be removed automatically.)

@sfc-gh-ema Please supplement if you have more specific examples.

Reply:
Write this as follows: 
if (param) {
old behavior
} else {
new behavior
}

Reply:
refactor it.